src/lib/PetitioRequest.ts

```diff
@@ -142,16 +142,49 @@ export class PetitioRequest {
 	 * set to the URL encoded version of the query string.
 	 */
 	public body(data: ParsedUrlQueryInput | string, sendAs: "form"): this
-	public body(data: any, sendAs?: "json" | "form"): this {
-		if (sendAs === "json" || (typeof data === "object" && !Buffer.isBuffer(data))) {
-			this.reqHeaders["content-type"] = "application/json";
-			this.data = JSON.stringify(data);
-		} else if (sendAs === "form") {
-			this.reqHeaders["content-type"] = "application/x-www-form-urlencoded";
-			this.data = stringify(data);
-		} else this.data = data;
-		this.reqHeaders["content-length"] = Buffer.byteLength(this.data as string | Buffer).toString();
-
+	/**
+	 * @param {*} data The data to be set for the request body.
+	 * @param {*} sendAs If data is a stream *AND* this is set to `stream`,
+	 * the body will be sent as is with no modifications such as stringification
+	 * or otherwise.
+	 */
+	public body(data: Readable, sendAs: "stream"): this
+	public body(data: any, sendAs?: "json" | "form" | "stream"): this {
+		switch (sendAs) {
+			case "json": {
+				this.data = JSON.stringify(data);
+				this.header({
+					"content-type": "application/json",
+					"content-length": Buffer.byteLength(this.data as string | Buffer).toString()
+				});
+				break;
+			}
+			case "form": {
+				this.data = stringify(data);
+				this.header({
+					"content-type": "application/x-www-form-urlencoded",
+					"content-length": Buffer.byteLength(this.data as string | Buffer).toString()
+				});
+				break;
+			}
```

Comment:
```suggestion
      case "form":
			case "json": {
				this.data = sendAs === 'json' ? JSON.stringify(data) : stringify(data);
				this.header({
					"content-type": `application/${sendAs === 'json' ? 'json' : 'x-www-form-urlencoded'}`,
					"content-length": Buffer.byteLength(this.data as string | Buffer).toString()
				});
				break;
			}
``` 
maybe this would speed up, I have no idea